### PR TITLE
Feature/ #636 Module Specific Readme Files

### PR DESCRIPTION
**Pull request by 10upsimon**
## Summary

Addresses #636 

## Relevant technical choices

- Adds a draft v1 `readme.txt` file to the `webp-uploads` module folder for copying over to the relevant build directory as per the work carried out in #662 .
- Flags all `readme.txt` files in all performance plugin module folders for `export-ignore` via `.gitattributes` glob pattern match.

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

### CODE REVIEW

File: .gitattributes
Comment (felixarntz) [inline .gitattributes]: The requirements state to only add a readme for WebP Uploads as part of this issue. Let's stick to that and ignore all other modules.

**Review by felixarntz — COMMENTED** [review]
@10upsimon Can you please limit the changes here to just the WebP Uploads module? Other modules to deploy as standalone plugins will be handled in a separate issue.

File: .gitattributes
Comment (10upsimon) [reply .gitattributes]: @felixarntz  I've changed this to a single 1 line glob pattern now, based on @joemcgill 's feedback so that we can match current and future `readme.txt` files

```
/modules/**/readme.txt export-ignore
```

Let me know if this is good, or if you want me to specifically define it as webp-uploads only, and single out modules in the future.

**Comment by 10upsimon** [comment]
@felixarntz I've addressed your comment RE restricting this to `webp-uploads` only, however I did implement a catch-all glob pattern match for all module `readme.txt` files in order to `export-ignore` them.

Once @mukeshpanchal27 's PR at #662 is approved and merged, I'll enhance the CLI command to handle the copy of the `readme.txt` files.

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt]: I don't think we should mention the Performance Lab plugin here given that this is the description for the standalone plugin. I think we can use the description that is also present in the module header:
```suggestion
Creates WebP versions for new JPEG image uploads if supported by the server.
```

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt]: The latter is not correct, this plugin _will_ function as a standalone plugin. That's part of why we're breaking them apart.

I would avoid the mention of the Performance Lab plugin here and focus purely on a brief description of what this plugin does.

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt]: The plugin should be called like the module, "WebP Uploads". It does not require Performance Lab, so shouldn't look as if it was an extension.

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt]: See above:
```suggestion
= Installation from within WordPress =

1. Visit **Plugins > Add New**.
2. Search for **WebP Uploads**.
3. Install and activate the **WebP Uploads** plugin.

= Manual installation =

1. Upload the entire `webp-uploads` folder to the `/wp-content/plugins/` directory.
2. Visit **Plugins**.
3. Activate the **WebP Uploads** plugin.
```

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt]: This should be removed, as the plugin will work as soon as it's activated.

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt]: I think we can remove this.

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt]: This refers to a module, which it shouldn't. The standalone plugin will have no notion of what is experimental or not.

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt]:
```suggestion
Feedback is encouraged and much appreciated, especially since this plugin may contain future WordPress core features. If you have suggestions or requests for new features, you can [submit them as an issue in the WordPress Performance Team's GitHub repository](https://github.com/WordPress/performance/issues/new/choose). If you need help with troubleshooting or have a question about the plugin, please [create a new topic on our support forum](https://wordpress.org/support/plugin/webp-uploads/#new-topic-0).
```

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt:52]: Can you add (slightly modified versions of) the two WebP Uploads specific entries in our current Performance Lab `readme.txt` here? Those are relevant and should be part of this readme. The mentioning of activating the module in Performance Lab should be removed from them though, as just by activating this plugin on its own those two questions may be relevant.

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt]: We typically refer to PRs in the changelog instead of issues.
```suggestion
* Initial release of the WebP Uploads plugin as a standalone plugin. ([664](https://github.com/WordPress/performance/pull/664))
```

**Review by felixarntz — COMMENTED** [review]
@10upsimon Thank you for the updates. I think there is a misunderstanding here, as the standalone plugins will be in fact standalone. They should not be branded/marketed as extensions to the Performance Lab plugin.

Right now, as part of Milestone 1 (see #656), they will exist in both places, as a standalone plugin and as a module in the Performance Lab plugin. But once we're done with Milestone 2, they will only exist as standalone plugin.

Using Performance Lab alongside them is perfectly fine and we would still encourage users to use Performance Lab if they want to follow along what the performance team is working on. But someone that wants to just use WebP would be perfectly fine to only install this plugin  on its own.

Regarding the dependency on #662, I don't think we technically have to wait for that PR. Once this PR is finalized, it can be merged as is, since #662 will copy the entire directory already, so the `readme.txt` here will be included.

File: modules/images/webp-uploads/readme.txt
Comment (mukeshpanchal27) [inline modules/images/webp-uploads/readme.txt]:
```suggestion
Tags:              performance, images, webp
```

I don't think we have to add `site health` and `measurement` for WebP Uploads

**Review by mukeshpanchal27 — COMMENTED** [review]
Thanks @10upsimon, I left one minor suggestion, and could you please remove the `readme.txt` file for sqlite?

**Comment by 10upsimon** [comment]
@mukeshpanchal27 @felixarntz many thanks for your feedback and comments. I believe I have addressed them all and have requested a re-review of the PR. I confirm the readme.txt was copied across as expected with @mukeshpanchal27 latest merge of #662  @felixarntz

File: modules/images/webp-uploads/readme.txt
Comment (mukeshpanchal27) [inline modules/images/webp-uploads/readme.txt:17]:
```suggestion
```
Remove extra line.

**Review by mukeshpanchal27 — APPROVED** [review]
Thanks @10upsimon, The changes look good to me.

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt]: Technically not a blocker, but since this is pretty much the only module / plugin that has a UI to it, we still may want to provide at least a bit more explanation in 2-3 short paragraphs. Things to potentially highlight:
* WebP images will be generated only for new uploads
* By default, only WebP images will be generated for JPEG uploads, only the original uploaded file will still exist as JPEG
* There is a checkbox in `Settings > Media` that can change the behavior though to generate _both_ JPEG and WebP for every sub-size (noting again that this will only affect newly uploaded images, i.e. after making that change)

File: modules/images/webp-uploads/readme.txt
Comment (felixarntz) [inline modules/images/webp-uploads/readme.txt]: No need for an extra "heading" here
```suggestion
= 1.0.0 =
```

**Review by felixarntz — COMMENTED** [review]
@10upsimon This is pretty much good to go now, just one small thing and one slightly bigger thing. It would be great if you could expand the description just a tiny bit to explain a bit more clearly what the plugin does.

We can also iterate in a follow up if you prefer for just that content, in order to unblock this PR. Let me know what you prefer.

File: modules/images/webp-uploads/readme.txt
Comment (10upsimon) [reply modules/images/webp-uploads/readme.txt]: @felixarntz (cc @mukeshpanchal27 ) this is ready for re-review following changes pushed in recent commit(s).

File: modules/images/webp-uploads/readme.txt
Comment (mukeshpanchal27) [inline modules/images/webp-uploads/readme.txt]:
```suggestion
This plugin adds WebP support for media uploads within the WordPress application. WebP images will be generated only for new uploads, pre-existing imagery will not be converted to WebP format. By default, WebP images will only be generated for JPEG uploads, only the original uploaded file will still exist as a JPEG image. All generated image sizes will exist as WebP only. If you wish to change this behaviour, there is a checkbox in `Settings > Media` that - when checked - will alter the behaviour of this plugin to generate both JPEG and WebP images for every sub-size (noting again that this will only affect newly uploaded images, i.e. after making said change).
```

Comment (10upsimon) [reply modules/images/webp-uploads/readme.txt]: Thanks @mukeshpanchal27 , updated

**Review by felixarntz — APPROVED** [review]
@10upsimon Thank you for the updates, LGTM!

**Pull request merged**
